File: cloud-collect-core/src/main/java/com/windea/demo/cloudcollect/core/domain/model/DataSchema.java
package com.windea.demo.cloudcollect.core.domain.model;

import com.windea.demo.cloudcollect.core.domain.entity.Collect;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

/**
 * 数据约束。用于导入/导出数据。
 */
@Data
@NoArgsConstructor
public class DataSchema implements Serializable {
	private static final long serialVersionUID = 2929238521052792644L;

	private List<Collect> collectList = new ArrayList<>();


	public DataSchema(List<Collect> collectList) {
		this.collectList = collectList;
	}
}

File: cloud-collect-core/src/main/kotlin/com/windea/demo/cloudcollect/core/domain/model/DataSchema.kt
package com.windea.demo.cloudcollect.core.domain.model

import com.windea.demo.cloudcollect.core.domain.enums.*

/**数据概要。用于导入导出数据。*/
data class DataSchema(
	val collectSchemaList: List<CollectSchema>
)

/**收藏的概要。用于导入导出数据。*/
data class CollectSchema(
	val name: String,
	val summary: String,
	val url: String,
	val logoUrl: String,
	val categoryName: String,
	val tagNames: List<String>,
	val type: CollectType
)
